framework/src/problems/EigenProblem.C

```diff
@@ -368,10 +391,7 @@ EigenProblem::computeResidualL2Norm()
   _nl_eigen->residualVectorBX() *= eigenvalue;
 
   // Compute entire residual
-  if (_negative_sign_eigen_kernel)
-    _nl_eigen->residualVectorAX() += _nl_eigen->residualVectorBX();
-  else
-    _nl_eigen->residualVectorAX() -= _nl_eigen->residualVectorBX();
+  _nl_eigen->residualVectorAX() -= _nl_eigen->residualVectorBX();
```

Comment:
I think I would prefer what is implemented here to be `+=` as that is consistent with a classic nonlinear solver

Reply:
This is because the above call to `computeResidualAB` removes the minus sign in Bx vector. We could call `computeResidualTags` directly here and then can `+=` here.

The encapsulation of the solver implementation with SLEPc is not good imho, like pre-scaling, post-scaling, separating power iteration and newton iteration outside, this sign treatment, etc.